src/file/mod.rs

```diff
@@ -686,6 +690,9 @@ impl NamedTempFile {
     /// secure in the in all cases. Please read the security section on the top
     /// level documentation of this type for details.
     ///
+    /// When calling this method on Unix systems, the file keeps the `600` permissions
```

Comment:
I'd add a new section for this, right above security.

Also, while technically correct, this statement is misleading. We avoid setting permissions because we don't really know _what_ to do:

1. If we set them _first_, we could end up setting permissive permissions in some globally accessible directory.
2. If we set them afterwards, the file will temporarily have the wrong permissions.
3. We don't actually _know_ what permissions the user wants.

I think it's safest to just state that the file's permissions will be _preserved_ and that the temporary file's default permissions are 0600.